ci/ci

```diff
@@ -0,0 +1,344 @@
+                bad_images = (bad_images.split("'")[1]).replace("\\n", " ")
+                print("The following untagged images will be removed -> " + bad_images)
+                cmd = 'docker rmi ' + bad_images
+                run_cmd(cmd)
+        except Exception as e:
+            log_error(e)
+
+
+def delete_older_image(bad_images):
+    if val.find('_ci_pbs') == -1:
+        bad_images = val.replace(':', '_')
+        bad_images += '_ci_pbs'
+    cmd = "docker images -q " + bad_images
+    bad_images = run_cmd(cmd,True)
+    if bad_images == "b''":
+        return
+    bad_images = (bad_images.split("'")[1]).replace("\\n", " ")
+    log_info("The following untagged images will be removed -> " + bad_images)
+    cmd = 'docker rmi ' + bad_images
+    run_cmd(cmd)
+
+
+def find_ci():
+    cur_dir = os.path.dirname(os.path.abspath(__file__))
+    return os.path.dirname(cur_dir)
+
+
+def tail_file(build_log_path):
+    with open(build_log_path, 'rb') as f:
+        f.seek(-2, os.SEEK_END)
+        while f.read(1) != b'\n':
+            f.seek(-2, os.SEEK_CUR)
+        return f.readline().decode()
+
+
+
+def check_for_existing_image():
+    env_dir = os.path.join(pbs_dirname, '.env')
+    val = read_from_file(env_dir)
+    if val != '':
+        val = val.split('=')[1]
+        if val.find('_ci_pbs') != -1:
+            val = val[:-7].replace('_', ':')
+    else:
+        val = 'centos:7'
+    search_str = val.replace(":", "_")
+    search_str += '_ci_pbs'
+    cmd = 'docker images -q ' + search_str
+    search_result = run_cmd(cmd,True)
+    if search_result != "b''":
+        write_to_file(env_dir, "CONTAINER_PLATFORM="+search_str)
+        return True
+    else:
+        write_to_file(env_dir, "CONTAINER_PLATFORM="+val)
+        return False
+
+
+def check_prerequisites():
+    # This function will check wether docker docker-compose commands are available
+    # or a  container is running that has pbs
+    cmd = "where" if platform.system() == "Windows" else "which"
+
+    try:
+        subprocess.run([cmd, "docker"], stdout=subprocess.DEVNULL)
+    except:
+        log_error("docker not found in PATH")
+
+    try:
+        subprocess.run([cmd, "docker-compose"], stdout=subprocess.DEVNULL)
+    except:
+        log_error("docker-compose not found in PATH")
+
+    try:
+        docker_container_process = subprocess.run(
+            ["docker-compose", "ps", "pbs-build"], stdout=subprocess.PIPE)
+        docker_container = str(docker_container_process.stdout)
+        if docker_container.find('Up') == -1:
+            log_error("No running service found")
+            try:
+                log_info('Attempting to start container')
+                os.chdir(pbs_dirname)
+                subprocess.run(["docker-compose", "down"],
+                               stdout=subprocess.DEVNULL)
+                image_check = check_for_existing_image()
+                subprocess.run(["docker-compose", "up", "-d"])
```

Comment:
we should run only if image_check is True

Reply:
No we need not do that, if image exists it will use that image and we wouldn't need to build dependencies, 
Image check returning False only means to build dependencies, not to stop the services all together.

Reply:
you are calling check_for_existing_image(), starting the containers and then checking the value of image_check.
is there a reason to do that?

Reply:
No i guess if you are asking if check image can be called after 197, then yes. it would work too.